data-8/jupyterhub-k8s
# Autoscaling for nodes (working with Google Cloud Platform)

Closes #74 

Summary
-------
Provide auto-scaling without killing user pods, based on Kubernete's `Unschedulable` tag for nodes. See scale/README.md for expected scaling behaviors.

How to test
-------
See README.md for directions. Please make sure `kubectl proxy` is ready before running `scale.py`

Comments requested
------
How do you want the scaling script be run? Some options:
1. Event driven - run as a pod and receive HTTP requests from cull, hub, etc
2. Running on provisioner - with cron or add python script to run again

How to set the parameters? Including `min utilization`, `max utilization`, etc

Is it safe to kill a node running cull on it? General proposals for other `critical service` to be protected?

What tests do you want to run with it? We have run some tests but far from comprehensive.
